activerecord/test/current_page_test.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require "bundler/inline"
+
+
+require "rack/test"
+require "action_controller/railtie"
+require "minitest/autorun"
+require "action_dispatch/journey"
+
+class BugTest < ActiveSupport::TestCase
+  attr_accessor :controller, :request
+
+  routes = ActionDispatch::Routing::RouteSet.new
+  routes.draw do
+    root "posts#index"
+    resources :posts, only: :index
+  end
+
+  include ActionView::Helpers::UrlHelper
+  include routes.url_helpers
+
+  def request_for_url(url, opts = {})
+    env = Rack::MockRequest.env_for("http://www.example.com#{url}", opts)
+    ActionDispatch::Request.new(env)
+  end
+
+  def test_returns_success
+    @request = request_for_url("/posts", method: :head)
+    assert current_page?(controller: "posts", action: "index")
+  end
+end
```

Comment:
I think you probably want to add a test to the existing `current_page?` tests in ActionView: https://github.com/rails/rails/blob/64c8340bcbfcb3c91c1bf60c4b3de84c9d577bbe/actionview/test/template/url_helper_test.rb#L821

Reply:
See comment below